viper/parser/expr.py

```diff
@@ -78,7 +78,7 @@ def get_expr(self):
     def number(self):
         orignum = get_original_if_0x_prefixed(self.expr, self.context)
         if orignum is None and isinstance(self.expr.n, int):
-            if not (-2**127 + 1 <= self.expr.n <= 2**127 - 1):
```

Comment:
@fubuloubu @DavidKnott I believe using the clamp values here as well for literals is correct?

Reply:
@daejunpark  please use MemoryPositions.MINNUM and MemoryPositions.MAXNUM. As we are really striving to use constant from now on ;)

Reply:
Not sure I understand the comment, but the low end should just be `-2**127`.

Int N is always defined as the range [-2**(N-1), 2**(N-1)-1]. One bit goes towards the sign bit (hence N-1), and you can think of the non-negative end of the interval additionally containing the 0 value, which is why the non-negative side is always one less in value than the negative side (both contain equal numbers of entries that way). Think of it as negative and non-negative (literally sign bit or no sign bit) vs. positive/negative. I think that's where people get confused.

Reply:
@fubuloubu yes I think that's what we have here:

Reply:
Yeah, I love the collections of constants. I wish we used more constants instead of literals everywhere.

If your comment about was to just use these constants, then I 100% agree.

Reply:
@daejunpark just use the  MINNUM / MAXNUM constants thanks. And thanks for finding it ;)

Reply:
@jacqueswww thanks for the suggestion. I fixed it.